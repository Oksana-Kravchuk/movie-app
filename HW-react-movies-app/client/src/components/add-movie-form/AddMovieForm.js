import React from 'react';
import { DatePicker } from 'antd';
import { Form, Formik } from 'formik';

import MultiSelect from '../multi-select';
import FormField from '../form-field';
import MovieFormValidation from '../../utils/form-validation';

const AddMovieForm = ({submitForm}) => {
  return (
    <Formik
      initialValues={{
        title: '',
        release_date: '',
        poster_path: '',
        overview: '',
        genres: [],
        runtime: '',
      }}
      validationSchema={MovieFormValidation}
      onSubmit={(values, actions) => {
        () => submitForm(values);
        actions.resetForm();
      }}
    >
      {({
        values,
        resetForm,
        setFieldValue,
        errors,
        touched,
        setFieldTouched,
      }) => (
        <Form className="form">
          <FormField
            id="title"
            name="title"
            type="text"
            label="First Tile"
            placeholder="Title"
            onChange={(val) => setFieldValue('title', val)}
          />
          <div className="form__row">
            {errors.release_date && touched.release_date ? (
              <div className="form__error">{errors.release_date}</div>
            ) : null}
            <label className="form__label">
              Release date
              <DatePicker
                defaultValue={values.release_date}
                name="release_date"
                placeholder="Select date"
                onBlur={() => setFieldTouched('release_date', true)}
                onChange={(_, dateString) =>
                  setFieldValue('release_date', dateString)
                }
              />
            </label>
          </div>
          <FormField
            name="poster_path"
            type="text"
            label="Movie url"
            placeholder="Movie url here"
          />
          <div className="form__row">
            {errors.genres && touched.genres ? (
              <div className="form__error">{errors.genres}</div>
            ) : null}
            <label className="form__label">
              Genre
              <MultiSelect
                defaultValue={values.genres}
                name="genres"
                placeholder="Select genre"
                onBlur={() => setFieldTouched('genres', true)}
                onChange={(option) => setFieldValue('genres', option)}
              />
            </label>
          </div>
          <FormField
            name="overview"
            type="text"
            label="Overview"
            placeholder="Overview here"
          />
          <FormField
            name="runtime"
            type="number"
            label="RunTime"
            placeholder="RunTime here"
          />
          <div className="form__buttons">
            <button
              onClick={resetForm}
              className="form__button button button--secondary"
              type="button"
            >
              Reset
            </button>
            <button type="submit" className="button button--primary">
              Submit
            </button>
          </div>
        </Form>
      )}
    </Formik>
  );
};

export default AddMovieForm;
