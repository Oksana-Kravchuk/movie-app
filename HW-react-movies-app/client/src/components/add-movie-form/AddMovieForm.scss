@import '../../common/styles/colours.scss';

.add-movie-form {
  button {
    padding: 18px 35px;
  }
}
