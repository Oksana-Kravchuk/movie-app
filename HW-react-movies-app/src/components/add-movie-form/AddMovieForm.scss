@import '../../common/styles/colours.scss';

.add-movie-form {
   button {
      padding: 18px 35px;
   }
}

.ant-picker input {
   color: $white;
   height: 40px;

   &::placeholder {
      color: $white;
      opacity: 0.4;
      letter-spacing: 1.4px;
   }
}

.ant-form-item {
   margin-bottom: 15px;;
}

.ant-input, .ant-picker, .ant-select-multiple {
   background-color: $gray-dark;
   height: 40px;
   letter-spacing: 1.4px;
   color: $white;
   width: 100%;
   border: none;

   &::placeholder {
      color: $white;
      opacity: 0.4;
   }
}

.ant-form-item-label > label {
   color: $primary;
   text-transform: uppercase;
   font-size: 12px;
}

.ant-picker-suffix {
   color: $primary;
   font-size: 16px;
}

.ant-select-single .ant-select-selector .ant-select-selection-item {
   line-height: unset;
}