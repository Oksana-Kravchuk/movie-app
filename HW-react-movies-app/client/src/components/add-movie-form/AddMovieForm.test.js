import React from 'react';
import { shallow } from 'enzyme';
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AddMovieForm from './AddMovieForm';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

describe('AddMovieForm', () => {
  const initialValues = {
    title: '',
    release_date: '',
    poster_path: '',
    overview: '',
    genres: [],
    runtime: '',
  };
  let wrapper;
  let resetFrom;
  let submitForm;
  
  beforeEach(() => {
    wrapper = shallow(<AddMovieForm {...initialValues} />);
    resetFrom = jest.fn();
    submitForm = jest.fn();
  });

  it('should match stored snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should display necessary form fields', () => {
    expect(wrapper.dive().find('FormField[name="title"]')).toBeTruthy();
    expect(wrapper.dive().find('FormField[name="poster_path"]')).toBeTruthy();
    expect(wrapper.dive().find('FormField[name="overview"]')).toBeTruthy();
    expect(wrapper.dive().find('FormField[name="runtime"]')).toBeTruthy();
    expect(wrapper.dive().find('MultiSelect[name="genres"]')).toBeTruthy();
    expect(wrapper.dive().find('DatePicker[name="release_date"]')).toBeTruthy();
  });

  it('should not submit if required fields are empty', () => {
    const button = wrapper.dive().find('button[type="submit"]');
    button.simulate('click', { preventDefault: () => {} });

    expect(submitForm).not.toHaveBeenCalled();
  });
});