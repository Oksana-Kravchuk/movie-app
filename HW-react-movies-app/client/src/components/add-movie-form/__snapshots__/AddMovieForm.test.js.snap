// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovieForm should match stored snapshot 1`] = `
<Formik
  initialValues={
    Object {
      "genres": Array [],
      "overview": "",
      "poster_path": "",
      "release_date": "",
      "runtime": "",
      "title": "",
    }
  }
  onSubmit={[Function]}
  validationSchema={
    ObjectSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_defaultDefault": [Function],
      "_deps": Array [],
      "_excludedEdges": Array [],
      "_exclusive": Object {},
      "_mutate": undefined,
      "_nodes": Array [
        "runtime",
        "overview",
        "genres",
        "poster_path",
        "release_date",
        "title",
      ],
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "object",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "fields": Object {
        "genres": ArraySchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "min": true,
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_subType": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {},
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "_type": "array",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "innerType": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {},
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "array",
        },
        "overview": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "poster_path": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "matches": false,
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "release_date": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "runtime": NumberSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "number",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "number",
        },
        "title": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
      "type": "object",
    }
  }
>
  <Component />
</Formik>
`;
